src/time.js

```diff
@@ -0,0 +1,85 @@
+        let minute   = split.slice(11, 13).join('')
+        let second   = split.slice(13, 15).join('')
+        let timezone = split.slice(15, 23).join('')
+        let date = new Date(`${year}-${month}-${day}T${hour}:${minute}:${second}.000${timezone}`)
+        return date.getTime() // Unix timestamp from Date object.
+    }
+
+    // Convert Date to TeamCity's weird time notation.
+    DateToTcTime(date) {
+        let year     = date.toISOString().substr(0, 4)
+        let month    = date.toISOString().substr(5, 2)
+        let day      = date.toISOString().substr(8, 2)
+        let hour     = '00' // Well... let's not get nitty gritty here.
+        let minute   = '00'
+        let second   = '00'
+        let timezone = '%2B0000' // +0000
+        let tcTime = `${year}${month}${day}T${hour}${minute}${second}${timezone}` // TeamCity time format: 20221206T080035+0100
+        return tcTime
+    }
+
+    // Convert HTML input datetime-local to TeamCity's weird time notation.
+    htmlDateTimeToTcTime(htmlDateTime) {
+        let split    = htmlDateTime.split('') // 2022-12-22T23:15
+        let year     = split.slice(0, 4).join('')
+        let month    = split.slice(5, 7).join('')
+        let day      = split.slice(8, 10).join('')
+        let t        = split.slice(10, 11).join('')
+        let hour     = split.slice(11, 13).join('')
+        let minute   = split.slice(14, 16).join('')
+        let second   = '00'
+        let timezone = '%2B0000' // +0000
+        let tcTime = `${year}${month}${day}T${hour}${minute}${second}${timezone}` // TeamCity time format: 20221206T080035+0100
+        return tcTime
+    }
+
+    // Convert TeamCity's weird time notation to Unix timestamp.
+    htmlDateTimeToUnix(htmlDateTime) {
+        let split    = htmlDateTime.split('') // 2022-12-22T23:15
+        let year     = split.slice(0, 4).join('')
+        let month    = split.slice(5, 7).join('')
+        let day      = split.slice(8, 10).join('')
+        let t        = split.slice(10, 11).join('')
+        let hour     = split.slice(11, 13).join('')
+        let minute   = split.slice(14, 16).join('')
+        let date = new Date(`${year}-${month}-${day}T${hour}:${minute}`)
+        return date.getTime() // Unix timestamp from Date object.
+    }
+
+    // Convert TeamCity's weird time notation to Unix timestamp.
+    htmlDateTimeToDate(htmlDateTime) {
+        let split    = htmlDateTime.split('') // 2022-12-22T23:15
+        let year     = split.slice(0, 4).join('')
+        let month    = split.slice(5, 7).join('')
+        let day      = split.slice(8, 10).join('')
+        let t        = split.slice(10, 11).join('')
+        let hour     = split.slice(11, 13).join('')
+        let minute   = split.slice(14, 16).join('')
+        return date = new Date(`${year}-${month}-${day}T${hour}:${minute}`)
+    }
+
+    // Cut-off date in TeamCity's weird time notation, used for API calls.
+    cutoffTcString = function (d) {
```

Comment:
Is there a reason why the last two methods in this file use the `name = function (...) {}` notation?

Reply:
No this was a module that I copied and forgot to take a closer look at. Will update it to conform to standard.